R/utils.R

```diff
@@ -67,9 +67,20 @@ is_url <- function(x) {
 
 # Given the path to a file, return a file:// URL.
 file_url <- function(filename) {
-  if (is_windows()) {
-    paste0("file://", normalizePath(filename, mustWork = TRUE))
+  if (is_windows() & getRversion() < R_system_version("4.2.0")) {
+    enc2 <- identify
   } else {
-    enc2utf8(paste0("file:///", normalizePath(filename, winslash = "/", mustWork = TRUE)))
+    enc2 <- enc2utf8
   }
+  file <- sub("(.*)#[^#]*", "\\1", filename)
+  anchor_suffix <- sub(file, "", filename)
```

Comment:
This regex probably needs to do more to ensure that its only capturing the hash after the file extension. On Mac, I created a file named `Your #1 fan!.html` and this resulted in

```r
filename <- "Your #1 fan!.html"
file <- sub("(.*)#[^#]*", "\\1", filename)
anchor_suffix <- sub(file, "", filename)

file
#> [1] "Your "
anchor_suffix
#> [1] "#1 fan!.html"
```

Reply:
The updated regex should allow more complex pattern.

```r
filename <- "Your #1 fan!.html?option#anchor"
file <- sub("^(.*)(\\.[^#]+)(#.*)?$", "\\1\\2", filename, perl = TRUE)
anchor_suffix <- sub(file, "", filename, fixed = TRUE)

file
#> [1] "Your #1 fan!.html?option"
anchor_suffix
#> [1] "#anchor"
```